client.go

```diff
@@ -111,6 +120,9 @@ func NewClient(c *Config) (*Client, error) {
 			return &cli, err
 		}
 		if cli.token.Renewable {
+			// Make sure the channel is created only when it's necessary
+			// (when token is renewable and there are no errors).
+			cli.done = make(chan bool)
```

Comment:
Tools like `os.Signal` require a channel that has a single buffer to ensure immediate and proper handling. Would we benefit from have a single buffer here as well?

Reply:
If we don't buffer, the call to `Shutdown` will block until `case <-c.done:` can read from the channel.
The delay would happen if the goroutine is in the middle of handling `case <-ticker.C:`, doing the http requests and stuff.
How big that delay would be, plus how worrisome that is, I am not sure

Reply:
Considering that the shutdown will be called in a hot path during authentication of each incoming request, I believe it would be safer to buffer this channel so exit is immediate